exodus_lambda/functions/verify_source.py

```diff
@@ -0,0 +1,56 @@
+import json
+import os
+import subprocess
+
+import boto3
+
+
+def lambda_handler(event, context):
+    """Verifies the commit delivered to the pipeline by GitHub has a
+    tag signed by a trusted entity.
+
+    Receives the commit ID from the pipeline via UserParameters.
+
+    This function must be manually deployed prior to pipeline creation.
+    """
+    # pylint: disable=unused-argument
+
+    job_id = event["CodePipeline.job"]["id"]
+    job_data = event["CodePipeline.job"]["data"]
+    job_params = json.loads(
+        job_data["actionConfiguration"]["configuration"]["UserParameters"]
+    )
+
+    pipeline_client = boto3.client("codepipeline")
+
+    try:
+        # Import whitelisted GPG keys
+        pub_keys = "".join(
```

Comment:
Is "".join right or does it have to be "\n".join ?

In the YAML file for example you have:

```
                                   -----END PGP PUBLIC KEY BLOCK-----",
                      "ngillett": "-----BEGIN PGP PUBLIC KEY BLOCK-----
```

Note the keys don't end with a newline.  If you joined those items then wouldn't you end up with a line like:

```
-----END PGP PUBLIC KEY BLOCK----------BEGIN PGP PUBLIC KEY BLOCK-----
```

Not sure if the format is documented but I'd have assumed that newlines were significant at least around the key begin/end delimiters.